pkg/daemon/daemon.go

```diff
@@ -1863,6 +1892,20 @@ func (dn *Daemon) checkOS(osImageURL string) bool {
 		return true
 	}
 
+	// TODO(jkyros): the header for this functions says "if the digests match"
+	// so I'm wondering if at one point this used to work this way....
+	// TODO(jkyros): and pulling it just to check is so expensive, skopeo works now
```

Comment:
It makes sense to switch to use skopeo and remove local maintained code of imageInspect if tags are no longer a concern. Will be good to track this in a card and make this change.